service/storage_service.cc

```diff
@@ -2756,15 +2757,51 @@ storage_service::prepare_replacement_info(std::unordered_set<gms::inet_address>
     if (!replace_host_id) {
         replace_host_id = _gossiper.get_host_id(replace_address);
     }
-    slogger.info("Host {}/{} is replacing {}/{}", get_token_metadata().get_my_id(), get_broadcast_address(), replace_host_id, replace_address);
-    co_await _gossiper.reset_endpoint_state_map();
 
-    co_return replacement_info {
+    auto ri = replacement_info {
         .tokens = std::move(tokens),
         .dc_rack = std::move(dc_rack),
         .host_id = std::move(replace_host_id),
         .address = replace_address,
     };
+
+    auto ignore_hioas = parse_node_list(_db.local().get_config().ignore_dead_nodes_for_replace());
```

Comment:
what is a hioas?

Reply:
It was supposed to be related to host_id and/or endpoint (or address), but it's a bad name
and I'm changing it in the next version.
Essentially I'm simplifying host_id_or_endpoint to keep a std::variant value that can be either host_id or endpoint, but not both.